module/applications/srd-compendium.mjs

```diff
@@ -31,23 +31,16 @@ export default class SRDCompendium extends Compendium {
     const documents = await this.collection.getDocuments();
     data.disclaimer = this.collection.get(this.constructor._SPECIAL_PAGES.disclaimer).pages.contents[0].text.content;
     data.chapters = documents.reduce((arr, entry) => {
-      const isChapter = entry.name.startsWith("Chapter");
-      const isAppendix = entry.name.startsWith("Appendix");
-      if ( !isChapter && !isAppendix ) return arr;
+      const type = entry.getFlag("dnd5e", "type");
+      if ( !type ) return arr;
       const e = entry.toObject();
-      e.showPages = (e.pages.length > 1) && isChapter;
+      e.showPages = (e.pages.length > 1) && (type === "chapter");
       arr.push(e);
       return arr;
     }, []).sort((a, b) => {
-      const [, chA] = a.name.match(/^Chapter (\d+)/) || [];
-      const [, chB] = b.name.match(/^Chapter (\d+)/) || [];
-      const [, appA] = a.name.match(/^Appendix ([A-Z])/) || [];
-      const [, appB] = b.name.match(/^Appendix ([A-Z])/) || [];
-      if ( chA && !chB ) return -1;
-      else if ( chB && !chA ) return 1;
-      if ( chA && chB ) return Number(chA) - Number(chB);
-      if ( appA && appB ) return appA.charCodeAt(0) - appB.charCodeAt(0);
-      return 0;
+      const positions = { chapter: 0, appendix: 100 };
+      const getOrder = o => (positions[o.flags?.dnd5e?.type] ?? 200) + (o.flags?.dnd5e?.position ?? 0);
+      return getOrder(a) - getOrder(b);
     });
```

Comment:
`getOrder` could probably be moved outside since it doesn't need to close over any of the variables inside the `sort` call.

```js
const getOrder = o => {
  return ({chapter: 0, appendix: 100}[o.flags?.dnd5e?.type] ?? 200) + (o.flags?.dnd5e?.position ?? 0);
};
```

Then `sort` can become:
```js
.sort((a, b) => getOrder(a) - getOrder(b));
```